custom_components/intesisbox/intesisbox.py

```diff
@@ -244,17 +268,37 @@ def set_horizontal_vane(self, vane: str):
     def _set_value(self, uid: str, value: str | int) -> None:
         """Change a setting on the thermostat."""
         try:
-            self._write(f"SET,1:{uid},{value}")
+            # self._write(f"SET,1:{uid},{value}")
+            asyncio.run(self._writeasync(f"SET,1:{uid},{value}"))
         except Exception as e:
             _LOGGER.error("%s Exception. %s / %s", type(e), e.args, e)
 
-    def set_mode(self, mode: str) -> None:
-        """Change the thermostat mode (heat, cool, etc)."""
-        if not self.is_on:
-            self.set_power_on()
-
+    def set_mode(self, mode):
+        """Send mode and confirm change before turning on."""
+        """ Some units return responses out of order"""
+        _LOGGER.debug(f"Setting MODE to {mode}.")
         if mode in MODES:
             self._set_value(FUNCTION_MODE, mode)
+        if not self.is_on:
+            """ "Check to ensure in correct mode before turning on"""
+            retry = 30
+            while self.mode != mode and retry > 0:
+                _LOGGER.debug(
+                    f"Waiting for MODE to return {mode}, currently {str(self.mode)}"
+                )
+                _LOGGER.debug(f"Retry attempt = {retry}")
+                # asyncio.run(self.delay(1)) # SHANE
```

Comment:
With this commented out I think we'll spam the controller while we wait for it to update right?

Reply:
I don't think so because we send the GET,1:MODE with a delay of 1 second in `_writeasync`.